lib/Target/TfheRust/TfheRustEmitter.cpp

```diff
@@ -91,8 +91,16 @@ std::pair<graph::Graph<Operation *>, Operation *> getGraph(Operation *op) {
 SmallVector<Value> getCiphertextOperands(ValueRange inputs) {
   SmallVector<Value> vals;
   for (Value val : inputs) {
-    // TODO(#474): Generalize to any encrypted uint.
-    if (isa<tfhe_rust::EncryptedUInt3Type>(val.getType())) {
+    mlir::Type vtype = val.getType();
+    if (isa<EncryptedUInt2Type>(vtype) || isa<EncryptedUInt3Type>(vtype) ||
```

Comment:
Optionally, I wonder if it would be helpful to add a simple trait to these types and isa<TheTraitClass> instead of this series of checks.